src/Aspire.Hosting/Dashboard/DashboardServiceData.cs

```diff
@@ -39,6 +39,9 @@ public DashboardServiceData(
         {
             static GenericResourceSnapshot CreateResourceSnapshot(IResource resource, string resourceId, DateTime creationTimestamp, CustomResourceSnapshot snapshot)
             {
+                // Extract icon information from resource annotations
+                var iconAnnotation = resource.Annotations.OfType<ResourceIconAnnotation>().FirstOrDefault();
```

Comment:
@mitchdenny @davidfowl Is this correct place to set icon information? Or should it be on `CustomResourceSnapshot` and the annotation data is set to the snapshot somewhere?

If it should be different, please be specific about where and how.